pkg/templates/wildcardinrules/template.go

```diff
@@ -0,0 +1,74 @@
+package wildcardinrules
+
+import (
+	"fmt"
+
+	"golang.stackrox.io/kube-linter/pkg/check"
+	"golang.stackrox.io/kube-linter/pkg/config"
+	"golang.stackrox.io/kube-linter/pkg/diagnostic"
+	"golang.stackrox.io/kube-linter/pkg/extract"
+	"golang.stackrox.io/kube-linter/pkg/lintcontext"
+	"golang.stackrox.io/kube-linter/pkg/objectkinds"
+	"golang.stackrox.io/kube-linter/pkg/templates"
+	"golang.stackrox.io/kube-linter/pkg/templates/wildcardinrules/internal/params"
+	rbacV1 "k8s.io/api/rbac/v1"
+)
+
+var (
+	roleGVK        = rbacV1.SchemeGroupVersion.WithKind(objectkinds.Role)
+	clusterRoleGVK = rbacV1.SchemeGroupVersion.WithKind(objectkinds.ClusterRole)
+)
+
+func init() {
+	templates.Register(check.Template{
+		HumanName:   "Wildcard Use in Role and ClusterRole Rules",
+		Key:         "wildcard-in-rules",
+		Description: "Flag Roles and ClusterRoles that use wildcard * in rules",
+		SupportedObjectKinds: config.ObjectKindsDesc{
+			ObjectKinds: []string{
+				objectkinds.Role,
+				objectkinds.ClusterRole},
+		},
+		Parameters:             params.ParamDescs,
+		ParseAndValidateParams: params.ParseAndValidate,
+		Instantiate: params.WrapInstantiateFunc(func(_ params.Params) (check.Func, error) {
+			return func(lintCtx lintcontext.LintContext, object lintcontext.Object) []diagnostic.Diagnostic {
+				gvk := extract.GVK(object.K8sObject)
+				if gvk == roleGVK {
+					role, ok := object.K8sObject.(*rbacV1.Role)
+					if !ok {
+						return nil
+					}
+					return findWildCard(role.Rules)
+				}
+
+				if gvk == clusterRoleGVK {
+					crole, ok := object.K8sObject.(*rbacV1.ClusterRole)
+					if !ok {
+						return nil
+					}
+					return findWildCard(crole.Rules)
+				}
```

Comment:
Same comments as for the above template wrt the redundancy of the `== gvk` check and the casting

Reply:
updated as suggested